### ISSUE TITLE
CSV encode double-escapes escape codes like CR

### ISSUE THREAD

**Issue by karmajunkie** [beatrichartz/csv#70]
When encoding a line which contains a field with an embedded escape code, CSV escapes the backslash, so the output contains `\r\n` instead of a CRLF. This is an issue with double-quoted fields which are allowed to have CRLF's embedded.  See below:

```
iex(9)> IO.puts "foo\r\nbar"
foo
bar
:ok
iex(10)> [["foo\r\nbar", "foo", "bar"]] |> CSV.encode |> Enum.to_list
["\"foo\\r\\nbar\",foo,bar\r\n"]
```

I can work around it by regex removing the escaped backslash, but unless I'm missing something this is incorrect behavior.

**Comment by beatrichartz** [comment]
Thanks for raising this, will have a look at a fix soon.

**Comment by chingan-tsc** [comment]
Am having exactly the same issue and my temporary fix is something like `String.replace("\\n", "\n")`.

**Comment by beatrichartz** [comment]
Hey all, this is fixed in `2.4.1` - feel free to reopen if issues persist.
